src/pkg/layout/split.go

```diff
@@ -18,30 +18,37 @@ import (
 )
 
 // splitFile will split the file into chunks and remove the original file.
-func splitFile(srcPath string, chunkSize int) error {
+func splitFile(srcPath string, chunkSize int) (err error) {
 	srcFile, err := os.Open(srcPath)
 	if err != nil {
 		return err
 	}
-	defer srcFile.Close()
 	fi, err := srcFile.Stat()
 	if err != nil {
 		return err
 	}
 
 	title := fmt.Sprintf("[0/%d] MB bytes written", fi.Size()/1000/1000)
 	progressBar := message.NewProgressBar(fi.Size(), title)
-	defer progressBar.Close()
+	defer func(progressBar *message.ProgressBar) {
+		err2 := progressBar.Close()
+		err = errors.Join(err, err2)
+	}(progressBar)
 
 	hash := sha256.New()
 	fileCount := 0
+	// TODO(mkcp): The inside of this loop should be wrapped in a closure so we can close the destination file each
+	//   iteration as soon as we're done writing.
 	for {
 		path := fmt.Sprintf("%s.part%03d", srcPath, fileCount+1)
 		dstFile, err := os.OpenFile(path, os.O_CREATE|os.O_RDWR, helpers.ReadAllWriteUser)
 		if err != nil {
 			return err
 		}
-		defer dstFile.Close()
+		defer func(dstFile *os.File) {
+			err2 := dstFile.Close()
+			err = errors.Join(err, err2)
+		}(dstFile)
```

Comment:
This defer is good, but we close the file again below. I know that's the existing code, but could you delete that other close

Reply:
Yup -- missed the inline close. Deleted in latest

Reply:
This actually caused some issues on windows. We need to close the file before removing it, and that can lead to some resource conflicts between a happy path close-and-delete, and a deferred close that we want to clean up any error paths. I ensured that any deferred close won't return its error if it's `os.ErrClosed` to get around this.